File: freecodecamp/daily-code-challenge/2026-03-30.py
from datetime import datetime
import calendar

def get_due_date(date_str):
    date = datetime.strptime(date_str, "%Y-%m-%d")
    
    # Add 9 months
    new_month = date.month + 9
    new_year = date.year + (new_month - 1) // 12
    new_month = (new_month - 1) % 12 + 1
    
    # Get last day of target month
    last_day = calendar.monthrange(new_year, new_month)[1]
    
    # Adjust day if needed
    new_day = min(date.day, last_day)
    
    return f"{new_year:04d}-{new_month:02d}-{new_day:02d}"

File: freecodecamp/daily-code-challenge/2026-03-30.js
function getDueDate(dateStr) {
  const [year, month, day] = dateStr.split("-").map(Number);

  // Add 9 months
  let newMonth = month + 9;
  let newYear = year + Math.floor((newMonth - 1) / 12);
  newMonth = ((newMonth - 1) % 12) + 1;

  // Get last day of target month
  const lastDay = new Date(newYear, newMonth, 0).getDate();

  // Adjust day if needed
  const newDay = Math.min(day, lastDay);

  return `${newYear.toString().padStart(4, "0")}-${newMonth
    .toString()
    .padStart(2, "0")}-${newDay.toString().padStart(2, "0")}`;
}